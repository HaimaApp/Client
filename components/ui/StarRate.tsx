import { AntDesign } from "@expo/vector-icons";
import { View } from "react-native";

interface StarProps {
  starIndex: number;
  totalStar: number;
}

const StarRate: React.FC<StarProps> = ({ starIndex, totalStar }) => {
  return (
    <View className="flex-row justify-center items-center space-x-1">
      {/* Render indicator */}
      {Array.from({ length: totalStar }).map((_, index) => (
        <AntDesign
          key={index}
          name="star"
          size={18}
          color={starIndex >= index ? "#E6B8B8" : "#272525"}
        />
      ))}
    </View>
  );
};

export default StarRate;
