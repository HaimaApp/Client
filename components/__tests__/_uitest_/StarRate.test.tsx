import React from "react";
import { render } from "@testing-library/react-native";
import StarRate from "@/components/ui/StarRate"; // Adjust this import path as needed

describe("StarRate Component", () => {
  it("renders the correct number of stars based on totalStar", () => {
    const { toJSON } = render(
      <StarRate starIndex={2} totalStar={5} />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it("renders stars with the correct color based on starIndex", () => {
    const { toJSON } = render(
      <StarRate starIndex={3} totalStar={5} />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
