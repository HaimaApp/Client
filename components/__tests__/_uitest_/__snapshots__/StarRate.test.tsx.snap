// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRate Component renders stars with the correct color based on starIndex 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "justifyContent": "center",
      },
      {
        "alignItems": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#E6B8B8",
          "fontSize": 18,
        },
        undefined,
        {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
    testID="star-icon-0"
  >
    
  </Text>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#E6B8B8",
          "fontSize": 18,
        },
        [
          {
            "marginLeft": 4,
          },
        ],
        {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
    testID="star-icon-1"
  >
    
  </Text>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#E6B8B8",
          "fontSize": 18,
        },
        [
          {
            "marginLeft": 4,
          },
        ],
        {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
    testID="star-icon-2"
  >
    
  </Text>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#E6B8B8",
          "fontSize": 18,
        },
        [
          {
            "marginLeft": 4,
          },
        ],
        {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
    testID="star-icon-3"
  >
    
  </Text>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#272525",
          "fontSize": 18,
        },
        [
          {
            "marginLeft": 4,
          },
        ],
        {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
    testID="star-icon-4"
  >
    
  </Text>
</View>
`;

exports[`StarRate Component renders the correct number of stars based on totalStar 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "justifyContent": "center",
      },
      {
        "alignItems": "center",
      },
    ]
  }
>
  <Text />
  <Text />
  <Text />
  <Text />
  <Text />
</View>
`;
